tests/StacktraceTest.php

```diff
@@ -138,54 +138,47 @@ public function testAddFrameMarksAsInApp(): void
         $this->assertTrue($frames[1]->isInApp());
     }
 
-    public function testAddFrameReadsCodeFromShortFile(): void
+    /**
+     * @dataProvider contextDataProvider
+     */
+    public function testAddFrameWithDifferentContexts($fixture, $lineNumber, $contextLines, $preContextCount, $postContextCount)
     {
-        $fileContent = explode("\n", $this->getFixture('code/ShortFile.php'));
+        if (isset($contextLines)) {
+            $this->options->setContextLines($contextLines);
+        }
+
+        $fileContent = explode("\n", $this->getFixture($fixture));
         $stacktrace = new Stacktrace($this->options, $this->serializer, $this->representationSerializer);
 
-        $stacktrace->addFrame($this->getFixturePath('code/ShortFile.php'), 3, ['function' => '[unknown]']);
+        $stacktrace->addFrame($this->getFixturePath($fixture), $lineNumber, ['function' => '[unknown]']);
 
         $frames = $stacktrace->getFrames();
 
         $this->assertCount(1, $frames);
-        $this->assertCount(2, $frames[0]->getPreContext());
-        $this->assertCount(2, $frames[0]->getPostContext());
+        $this->assertCount($preContextCount, $frames[0]->getPreContext());
+        $this->assertCount($postContextCount, $frames[0]->getPostContext());
 
-        for ($i = 0; $i < 2; ++$i) {
-            $this->assertEquals(rtrim($fileContent[$i]), $frames[0]->getPreContext()[$i]);
+        for ($i = 0; $i < $preContextCount; ++$i) {
+            $this->assertEquals(rtrim($fileContent[$i + ($lineNumber - $preContextCount - 1)]), $frames[0]->getPreContext()[$i]);
         }
 
-        $this->assertEquals(rtrim($fileContent[2]), $frames[0]->getContextLine());
+        $this->assertEquals(rtrim($fileContent[$lineNumber - 1]), $frames[0]->getContextLine());
 
-        for ($i = 0; $i < 2; ++$i) {
-            $this->assertEquals(rtrim($fileContent[$i + 3]), $frames[0]->getPostContext()[$i]);
+        for ($i = 0; $i < $postContextCount; ++$i) {
+            $this->assertEquals(rtrim($fileContent[$i + $lineNumber]), $frames[0]->getPostContext()[$i]);
         }
     }
 
-    public function testAddFrameReadsCodeFromLongFile(): void
+    public function contextDataProvider()
```

Comment:
As a convention used across the codebase we used for the name of the method of the data provider the name of the test method that uses it + the `DataProvider` suffix. The only exception is if the data provider is used from more than one test method

Reply:
Done